pombola/kenya/views_facebook_experiments.py

```diff
@@ -0,0 +1,200 @@
+
+    # The survey doesn't actually care if this works or not, just send an "OK".
+    template_name = 'facebook-ajax-response.html'
+
+    def get(self, request, *args, **kwargs):
+        survey_session = request.GET.get('sid')
+        question_id = request.GET.get('qid')
+        selection_id = request.GET.get('selid')
+
+        if survey_session and question_id and selection_id:
+            recordPositiveResponse(survey_session, question_id, selection_id)
+
+        return super(SurveyGizmoRecord, self).get(request, *args, **kwargs)
+
+
+class SurveyGizmoValue(TemplateView):
+
+    # The survey doesn't actually care if this works or not, just send an "OK".
+    template_name = 'facebook-ajax-response.html'
+
+    def get(self, request, *args, **kwargs):
+        survey_session = request.GET.get('sid')
+        question_id = request.GET.get('qid')
+        value = request.GET.get('value')
+
+        if survey_session and question_id and value:
+            recordResponseValue(survey_session, question_id, value)
+
+        return super(SurveyGizmoValue, self).get(request, *args, **kwargs)
+
+
+def getSurveyIdFromSession(survey_session):
+
+    survey_id = cache.get(survey_session)
+
+    if not survey_id:
+
+        url = "https://restapi.surveygizmo.com/v4/survey/2802241/surveyresponse/"
+
+        querystring = {
+            "api_token": settings.SURVEYGIZMO_API_TOKEN,
+            "api_token_secret": settings.SURVEYGIZMO_API_SECRET,
+            "filter[field][0]": "[question(13)]",
+            "filter[operator][0]": "=",
+            "filter[value][0]": survey_session
+            }
+
+        headers = {
+            'cache-control': "no-cache"
+            }
+
+        response = requests.request("GET", url,
+                                    headers=headers,
+                                    params=querystring
+                                    )
+        data = json.loads(response.text)
+
+        # Sanity check we only have one response. If not, abandon ship.
+        if int(data['total_count']) == 1:
+            survey = data['data'][0]
+            survey_id = survey['id']
+
+        else:
+            survey_id = None
+
+        cache.set(survey_session, survey_id, 1800)
+
+        return survey_id
+
+
+def recordPositiveResponse(survey_session, question_id, selection_id):
+
+    # Get the survey ID
+    survey_id = getSurveyIdFromSession(survey_session)
+
+    # Sanity check that it's a real survey ID
+    if survey_id:
+
+        url = "https://restapi.surveygizmo.com/v4/survey/2802241/surveyresponse/" + survey_id + "/"
+
+        querystring = {
+            "api_token": settings.SURVEYGIZMO_API_TOKEN,
+            "api_token_secret": settings.SURVEYGIZMO_API_SECRET,
+            "data[" + question_id + "][" + selection_id + "]": "Yes"
+            }
+
+        headers = {
+            'cache-control': "no-cache",
+            }
+
+        response = requests.request("POST", url, headers=headers, params=querystring)
```

Comment:
`response` in this and the next function is unused; either the `response = ` should be dropped or it should be used to handle error responses (e.g. with `response.raise_for_status()`) - presumably it would be good to get an email if these POSTs are failing.